internal/venice-client-common/src/main/java/com/linkedin/venice/schema/avro/SchemaCompatibility19.java

```diff
@@ -0,0 +1,975 @@
+  private SchemaCompatibility19() {
+  }
+
+  /** Message to annotate reader/writer schema pairs that are compatible. */
+  public static final String READER_WRITER_COMPATIBLE_MESSAGE =
+      "Reader schema can always successfully decode data written using the writer schema.";
+
+  /**
+   * Validates that the provided reader schema can be used to decode avro data
+   * written with the provided writer schema.
+   *
+   * @param reader schema to check.
+   * @param writer schema to check.
+   * @return a result object identifying any compatibility errors.
+   */
+  public static SchemaPairCompatibility checkReaderWriterCompatibility(final Schema reader, final Schema writer) {
+    final SchemaCompatibilityResult compatibility =
+        new ReaderWriterCompatibilityChecker().getCompatibility(reader, writer);
+
+    final String message;
+    switch (compatibility.getCompatibility()) {
+      case INCOMPATIBLE: {
+        message = String.format(
+            "Data encoded using writer schema:%n%s%n" + "will or may fail to decode using reader schema:%n%s%n",
+            writer.toString(true),
+            reader.toString(true));
+        break;
+      }
+      case COMPATIBLE: {
+        message = READER_WRITER_COMPATIBLE_MESSAGE;
+        break;
+      }
+      default:
+        throw new AvroRuntimeException("Unknown compatibility: " + compatibility);
+    }
+
+    return new SchemaPairCompatibility(compatibility, reader, writer, message);
+  }
+
+  // -----------------------------------------------------------------------------------------------
+
+  /**
+   * Tests the equality of two Avro named schemas.
+   *
+   * <p>
+   * Matching includes reader name aliases.
+   * </p>
+   *
+   * @param reader Named reader schema.
+   * @param writer Named writer schema.
+   * @return whether the names of the named schemas match or not.
+   */
+  public static boolean schemaNameEquals(final Schema reader, final Schema writer) {
+    if (objectsEqual(reader.getName(), writer.getName())) {
+      return true;
+    }
+    // Apply reader aliases:
+    return reader.getAliases().contains(writer.getFullName());
+  }
+
+  /**
+   * Identifies the writer field that corresponds to the specified reader field.
+   *
+   * <p>
+   * Matching includes reader name aliases.
+   * </p>
+   *
+   * @param writerSchema Schema of the record where to look for the writer field.
+   * @param readerField  Reader field to identify the corresponding writer field
+   *                     of.
+   * @return the writer field, if any does correspond, or None.
+   */
+  public static Field lookupWriterField(final Schema writerSchema, final Field readerField) {
+    assert (writerSchema.getType() == Type.RECORD);
+    final List<Field> writerFields = new ArrayList<>();
+    final Field direct = writerSchema.getField(readerField.name());
+    if (direct != null) {
+      writerFields.add(direct);
+    }
+    for (final String readerFieldAliasName: readerField.aliases()) {
```

Comment:
Can we use `AvroCompatibilityHelper` for a few things in this class?

For example, we should use `AvroCompatibilityHelper.getFieldAliases(readerField)` here

Reply:
Based on the internal discussion, we will keep it as it is for now.